### PR TITLE
Improve source generation performance

### LINKED ISSUE #139: Android and iOS projects fail to reference netstandard2.0-only projects
Is it possible to add netStandard2 project references in the Droid project? 
I'm getting this error:
```
7>uno.sourcegenerationtasks\1.21.0\build\MonoAndroid\Uno.SourceGenerationTasks.targets(94,3): error : System.InvalidOperationException: The project(s) ButchersQA.ViewModels,GlobeWeigh.Rules,GlobeWeigh.Mvvm,ButchersQA.WebClient,ButchersQA.ClientContracts did not provide any metadata reference. This may be due to an invalid path, such as $(SolutionDir) being used in the csproj; try using relative paths instead.This may also be related to a missing default configuration directive. Refer to the Uno.SourceGenerator Readme.md file for more details.
7>uno.sourcegenerationtasks\1.21.0\build\MonoAndroid\Uno.SourceGenerationTasks.targets(94,3): error :    at Uno.SourceGeneration.Host.SourceGeneratorHost.<GetCompilation>d__4.MoveNext() in C:\projects\uno-sourcegeneration\src\Uno.SourceGenerationHost.Shared\SourceGeneratorHost.cs:line 303
```

### PR DESCRIPTION
**Pull request by jeromelaban**
GitHub Issue (If applicable): fixes #139

> **Reviewers: You may skip the first commit, it contains the `NET46` -> `NET461` replace all.**

## PR Type
What kind of change does this PR introduce?
- Bugfix

## What is the current behavior?
Uno.UI solution build time is longer than it should be, cross targeted projects build are serialized.

## What is the new behavior?
- Move to the latest Uno.SourceGenerationTasks to improve project parsing performance, and allows for the removal of unused targets caused by https://github.com/nventive/Uno.SourceGeneration/issues/2. Uno.Xaml and Uno.UI.BindingHelpers now only build the required targets.
- Move to `net461` for test projects so the the `Uno.Xaml` project can be referenced properly
- Use the latest MSBuild.Sdk.Extras for actual parallel cross-targeted builds
- Move the nuget package versions to the [`Directory.Build.targets`](src/Directory.Build.targets) file so it's easier to change all versions at once.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] Tested code with current [supported SDKs](../README.md#supported)
- [ ] Docs have been added/updated which fit [documentation template](https://github.com/nventive/Uno/blob/master/doc/.feature-template.md). (for bug fixes / features)
- [x] [Unit Tests and/or UI Tests](doc/articles/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [x] Contains **NO** breaking changes
- [ ] Updated the [Release Notes](https://github.com/nventive/Uno/tree/master/doc/ReleaseNotes)
- [ ] Associated with an issue (GitHub or internal)

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: doc/ReleaseNotes/_ReleaseNotes.md [hunk 1]
```diff
@@ -72,6 +72,7 @@
 * [WASM] Improve element arrange and transform performance
 * Restore original SymbolIcon.SymbolProperty as a C# property
 * Add support for `MediaPlaybackList`
+* Update Uno.SourceGenerationTasks to improve build performance
```
Comment (MatFillion) [inline doc/ReleaseNotes/_ReleaseNotes.md]: I think we can add more here :) 
Your PR description of the new behavior was very good, maybe we could add it here as well ?
```
Move to the latest Uno.SourceGenerationTasks to improve project parsing performance, and allows for the removal of unused targets caused by nventive/Uno.SourceGeneration#2. Uno.Xaml and Uno.UI.BindingHelpers now only build the required targets.
Move to net461 for test projects so the the Uno.Xaml project can be referenced properly
Use the latest MSBuild.Sdk.Extras for actual parallel cross-targeted builds
Move the nuget package versions to the Directory.Build.targets file so it's easier to change all versions at once.
```

File: doc/ReleaseNotes/_ReleaseNotes.md [hunk 2]
```diff
@@ -148,6 +149,7 @@
  * 146648 [Android] fixed ListView grouped items corruption on scroll
  * [Wasm] Fix `ListView` recycling when the `XamlParent` is not available for `AutoSuggestBox`
  * 147405 Fix NRE on some MediaTransportControl controls
+ * #139 Update Uno.SourceGenerationTasks to improve build performance
```
Comment (MatFillion) [inline doc/ReleaseNotes/_ReleaseNotes.md]: What is the required action for a project using uno to adapt to this breaking change?

Comment (MatFillion) [reply doc/ReleaseNotes/_ReleaseNotes.md]: nevermind! thought this was under breaking changes !

File: doc/articles/debugging-uno-ui.md [hunk 3]
```diff
@@ -25,6 +25,12 @@ step into Uno.UI without downloading the repository.
 Make sure **Enable source link support** check box is checked in **Tools** / **Options**
 / **Debugging** / **General** properties page.
 
+## Updating the Nuget packages used by the Uno.UI solution
```
Comment (MatFillion) [inline doc/articles/debugging-uno-ui.md]: That is something we could and should automate with our nuget updater ? @benventive  What do you think?

File: src/Uno.CrossTargetting.props [hunk 4]
```diff
@@ -61,8 +61,8 @@
   </PropertyGroup>
 
   <PropertyGroup Condition=" '$(TargetFramework)' == 'uap10.0' ">
-	<TargetPlatformVersion>10.0.14393.0</TargetPlatformVersion>
-	<TargetPlatformMinVersion>10.0.14393.0</TargetPlatformMinVersion>
+	<TargetPlatformVersion>10.0.16299.0</TargetPlatformVersion>
```
Comment (MatFillion) [inline src/Uno.CrossTargetting.props]: shouldn't we go for the more current one? 17763.

See umbrella PRs : https://nventive.visualstudio.com/Umbrella/_git/Umbrella.View/pullrequest/19858?_a=files&path=%2Fbuild%2FUmbrella.View.CrossTargeting.props

File: doc/articles/debugging-uno-ui.md (hunk 3 above)
Comment (jeromelaban) [reply doc/articles/debugging-uno-ui.md]: The nuget updater should be picking those up, I don't think it's filtering on the `Include` attribute.

File: src/Uno.CrossTargetting.props (hunk 4 above)
Comment (jeromelaban) [reply src/Uno.CrossTargetting.props]: Right, let's sync.

File: doc/articles/debugging-uno-ui.md (hunk 3 above)
Comment (jeromelaban) [reply doc/articles/debugging-uno-ui.md]: Will be fixed in nuget updater.

**Pull request merged**